src/coordinator/storage/local/storage_test.go

```diff
@@ -123,16 +223,101 @@ func TestLocalRead(t *testing.T) {
 	assert.Equal(t, tags, results.SeriesList[0].Tags)
 }
 
-func setupLocalSearch(t *testing.T) storage.Storage {
+func TestLocalSearchError(t *testing.T) {
 	ctrl := gomock.NewController(t)
-	store, session := setup(ctrl)
-	session.EXPECT().FetchTaggedIDs(gomock.Any(), gomock.Any(), gomock.Any()).Return(nil, false, errors.ErrNotImplemented)
-	return store
-}
+	defer ctrl.Finish()
+	store, sessions := setup(t, ctrl)
+	sessions.forEach(func(session *client.MockSession) {
+		session.EXPECT().FetchTaggedIDs(gomock.Any(), gomock.Any(), gomock.Any()).
+			Return(nil, false, fmt.Errorf("an error"))
+	})
 
-func TestLocalSearchExpectedFail(t *testing.T) {
-	store := setupLocalSearch(t)
 	searchReq := newFetchReq()
 	_, err := store.FetchTags(context.TODO(), searchReq, &storage.FetchOptions{Limit: 100})
 	assert.Error(t, err)
 }
+
+func TestLocalSearchSuccess(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+	store, sessions := setup(t, ctrl)
+
+	type testFetchTaggedID struct {
+		id        string
+		namespace string
+		tagName   string
+		tagValue  string
+	}
+
+	fetches := []testFetchTaggedID{
+		{
+			id:        "foo",
+			namespace: "metrics_unaggregated",
+			tagName:   "qux",
+			tagValue:  "qaz",
+		},
+		{
+			id:        "bar",
+			namespace: "metrics_aggregated",
+			tagName:   "qel",
+			tagValue:  "quz",
+		},
+	}
+
+	sessions.forEach(func(session *client.MockSession) {
+		var f testFetchTaggedID
+		switch {
+		case session == sessions.unaggregated1MonthRetention:
+			f = fetches[0]
+		case session == sessions.aggregated1MonthRetention1MinuteResolution:
+			f = fetches[1]
+		default:
+			panic("unexpected session")
+		}
+		iter := client.NewMockTaggedIDsIterator(ctrl)
+		gomock.InOrder(
+			iter.EXPECT().Next().Return(true),
+			iter.EXPECT().Current().Return(
+				ident.StringID(f.namespace),
+				ident.StringID(f.id),
+				ident.NewTagsIterator(ident.NewTags(
+					ident.Tag{
+						Name:  ident.StringID(f.tagName),
+						Value: ident.StringID(f.tagValue),
+					})),
+			),
+			iter.EXPECT().Next().Return(false),
+			iter.EXPECT().Err().Return(nil),
+			iter.EXPECT().Finalize(),
+		)
+
+		session.EXPECT().FetchTaggedIDs(gomock.Any(), gomock.Any(), gomock.Any()).
+			Return(iter, true, nil)
+	})
+	searchReq := newFetchReq()
+	result, err := store.FetchTags(context.TODO(), searchReq, &storage.FetchOptions{Limit: 100})
+	require.NoError(t, err)
+
+	require.Equal(t, len(fetches), len(result.Metrics))
```

Comment:
nit: `require.Len(t, result.Metrics, len(fetches))`

Reply:
`require.Len(...)` does some weird things including spewing out weird output when the length is wrong rather than just "2 != 3", so I always avoid it now...

Reply:
Oh damn, been telling people to use `X.Len` instead of `Equal` in reviews... 👀

Reply:
It works fine until the test stops passing, and then you spend like a full minute or two wondering whats going on lol.